src/pages/[platform]/build-a-backend/data/connect-to-existing-data-sources/connect-postgres-mysql-database/index.mdx

```diff
@@ -0,0 +1,366 @@
+import { getCustomStaticPath } from '@/utils/getCustomStaticPath';
+
+export const meta = {
+  title: 'Connect your app to existing MySQL and PostgreSQL database.',
+  description:
+    'Learn how to connect your app to existing MySQL and PostgreSQL database.',
+  platforms: [
+    'android',
+    'angular',
+    'flutter',
+    'javascript',
+    'nextjs',
+    'react',
+    'react-native',
+    'swift',
+    'vue',
+  ],
+};
+
+export const getStaticPaths = async () => {
+  return getCustomStaticPath(meta.platforms);
+};
+
+export function getStaticProps() {
+  return {
+    props: {
+      meta
+    }
+  };
+}
+
+
+Amplify's native integration supports any MySQL or Postgres database, no matter if they're hosted on AWS within a VPC or outside of AWS with a 3rd party hosted database provider.
+
+You must create a connection string using the following database information to get started:
+
+<ul>
+<li>
+<input type="checkbox"/> Database **hostname**
+</li>
+<li>
+<input type="checkbox"/> Database **port**
+</li>
+<li>
+<input type="checkbox"/> Database **username**
+</li>
+<li>
+<input type="checkbox"/> Database **user password**
+</li>
+<li>
+<input type="checkbox"/> Database **name**
+</li>
+</ul>
+
+## Step 1 - Set secrets for database connection
+
+First, provide all the database connection information as secrets, you can use the Amplify sandbox's secret functionality to set them or go to the Amplify console to set secrets in a shared environment:
+
+```bash showLineNumbers={false}
+npx ampx sandbox secret set SQL_CONNECTION_STRING
+```
+
+<BlockSwitcher>
+<Block name="MySQL">
+**Connection string format for MySQL**
+```console showLineNumbers={false}
+mysql://user:password@hostname:port/db-name
+```
+</Block>
+
+<Block name="PostgreSQL">
+**Connection string format for PostgreSQL**
+```console showLineNumbers={false}
+postgres://user:password@hostname:port/db-name
+```
+</Block>
+</BlockSwitcher>
+
+
+## Step 2 - Generate TypeScript representation of your database schema
+
+Run the following command to generate the Data schema with your database connection information. It'll infer an `a.model()` representation for **all database tables that have primary key specified**.
+
+```bash
+npx ampx generate schema-from-database --connection-uri-secret SQL_CONNECTION_STRING --out amplify/data/schema.sql.ts
+```
+
+<Accordion eyebrow="Info" title="Connecting to a database behind the VPC">
+
+If your database is behind a VPC, we'll autodetect the VPC configuration based on your connection string. This happens via a hostname lookup within your account. Make sure that your VPC is configured correctly to allow for 
+
+</Accordion>
+
```

Comment:
@palpatim @phani-srikar @sundersc  - do you mind pushing a commit with the most up-to-date VPC requirements here.